### PR TITLE
[Impeller] started scaling the gaussian blur sigma to match skia output

### PR DESCRIPTION
**Pull request by gaaclarke**
fixes https://github.com/flutter/flutter/issues/127770

This will help performance as well since our kernels will be smaller.  The justification for scaling the sigma is that we need to account in the math the scaling down that we are doing of the input.

Here are the side-by-side comparisons with this math and skia, impeller is always on the right:

## sigma 10

blur_10

## sigma 75

blur_75

## sigma 164 (sigma in bug)
blur_164

## sigma 400
blur_400

## sigma 1000
blur_1000

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by gaaclarke** [comment]
I'm not sure what sort of test we should add.  This will probably be picked up in existing goldens.

File: impeller/entity/contents/filters/directional_gaussian_blur_filter_contents.cc [hunk 1]
```diff
@@ -23,6 +23,23 @@
...
+//   0 | 1
+//  75 | 0.8
+// 150 | 0.5
+// 300 | 0.22
+// 400 | 0.2
+// 500 | 0.15
+Sigma ScaleSigma(Sigma sigma) {
+  // Limit the kernel size to 1000x1000 pixels, like Skia does.
+  Scalar clamped = std::min(sigma.sigma, 500.0f);
+  Scalar scalar = 1.02 - 3.89e-3 * clamped + 4.36e-06 * clamped * clamped;
```
Comment (jonahwilliams) [inline impeller/entity/contents/filters/directional_gaussian_blur_filter_contents.cc:39]: TIL you can use scientific notation in C++

**Review by jonahwilliams — APPROVED** [review]
IIRC from reading the Skia code, it goes something like

- If the Sigma is <= 4, leave it alone.
- Otherwise scale the Sigma down to 4, and use that same scaling factor to adjust the texture size.

So ScaleSigma is related to the curve we use below to scale the texture size, as in the both need to approximately agree.

If this looks like an improvement overall, then LGTM, but we might end up adjusting the whole thing anyway

**Comment by gaaclarke** [comment]
Here is the difference between skia and impeller with sigma 164 before this change, just to show this is an obvious improvement.

Screenshot 2023-10-27 at 3 58 30 PM

**Comment by jonahwilliams** [comment]
Yeah that looks way better

**Review by bdero — APPROVED** [review]
Yes! Fix the look of your blur with this one line trick. :)

The scaledown purely _adds_ to the blurriness. Our blurs already look pretty good, so just remap it to match Skia as a holdover.

We could always adjust this later with a more closely fitted curve, but this parabola looks pretty close!

**Comment by gaaclarke** [comment]
I looked through the goldens and they all seemed kosher.  Since there is existing coverage, I think I'm going to avoid adding another golden test.  `ClippedBlurFilterRendersCorrectlyInteractive` does have a large sigma.

**Pull request merged**